### PR TITLE
MCR-1719 use choose to put either display name or name part in index

### PR DESCRIPTION
**Pull request by kkrebs**

### CODE REVIEW

**Comment by MyCoRe-Travis** [comment]
Something gone wrong on d778d1a46fa1c32b12245f39d3d9907a44cce6cb! 
 Please Check https://github.com/MyCoRe-Travis/test_artifacts/tree/2145-pull_request-391

File: mycore-mods/src/main/resources/xsl/mods-solr.xsl [hunk 1]
```diff
@@ -67,24 +67,45 @@
     <!-- keep mods:name fields for legacy reasons -->
     <xsl:for-each select=".//mods:name">
       <field name="mods.name">
-        <xsl:for-each select="mods:displayForm | mods:namePart[@type!='date'] | mods:namePart[not(@type)] | text()">
-          <xsl:value-of select="concat(' ',mcrxsl:normalizeUnicode(.))" />
-        </xsl:for-each>
+        <xsl:choose>
+          <xsl:when test="mods:displayForm">
+            <xsl:value-of select="concat(' ',mcrxsl:normalizeUnicode(mods:displayForm))" />
```
Comment (yagee-de) [inline mycore-mods/src/main/resources/xsl/mods-solr.xsl]: no concat needed here

File: mycore-mods/src/main/resources/xsl/mods-solr.xsl [hunk 2]
```diff
@@ -67,24 +67,45 @@
...
       </field>
     </xsl:for-each>
     <xsl:for-each select="mods:name">
       <field name="mods.name.top">
-        <xsl:for-each select="mods:displayForm | mods:namePart[@type!='date'] | text()">
-          <xsl:value-of select="concat(' ',mcrxsl:normalizeUnicode(.))" />
-        </xsl:for-each>
+        <xsl:choose>
+          <xsl:when test="mods:displayForm">
+            <xsl:value-of select="concat(' ',mcrxsl:normalizeUnicode(mods:displayForm))" />
```
Comment (yagee-de) [inline mycore-mods/src/main/resources/xsl/mods-solr.xsl]: same here: no 'concat' needed

File: mycore-mods/src/main/resources/xsl/mods-solr.xsl [hunk 3]
```diff
@@ -67,24 +67,45 @@
...
     </xsl:for-each>
     <xsl:for-each
       select=".//mods:name[mods:role/mods:roleTerm[@authority='marcrelator' and (@type='text' and text()='author') or (@type='code' and text()='aut')]]">
       <field name="mods.author">
-        <xsl:for-each select="mods:displayForm | mods:namePart | text()">
-          <xsl:value-of select="concat(' ',mcrxsl:normalizeUnicode(.))" />
-        </xsl:for-each>
+        <xsl:choose>
+          <xsl:when test="mods:displayForm">
+            <xsl:value-of select="concat(' ',mcrxsl:normalizeUnicode(mods:displayForm))" />
```
Comment (yagee-de) [inline mycore-mods/src/main/resources/xsl/mods-solr.xsl]: same here: no 'concat' needed

File: mycore-mods/src/main/resources/xsl/mods-solr.xsl [hunk 4]
```diff
@@ -95,9 +116,16 @@
     <xsl:for-each
       select=".//mods:originInfo[not(@eventType) or @eventType='publication']/mods:publisher">
       <field name="mods.publisher">
-        <xsl:for-each select="mods:displayForm | mods:namePart | text()">
-          <xsl:value-of select="concat(' ',.)" />
-        </xsl:for-each>
+        <xsl:choose>
+          <xsl:when test="mods:displayForm">
+            <xsl:value-of select="concat(' ',mcrxsl:normalizeUnicode(mods:displayForm))" />
```
Comment (yagee-de) [inline mycore-mods/src/main/resources/xsl/mods-solr.xsl]: same here: no 'concat' needed

probably mods:name/mods:publisher deserves a custom template that returns a value for an index field

**Review by yagee-de — CHANGES_REQUESTED** [review]

File: mycore-mods/src/main/resources/xsl/mods-solr.xsl (hunk 4 above)
Comment (kkrebs) [reply mycore-mods/src/main/resources/xsl/mods-solr.xsl]: I took a look in my index and I think there is no need for this. We use a standard solr textfield here with tokenizer.

**Review by yagee-de — APPROVED** [review]

**Pull request merged**